packages/runtime/src/runtime/navigateApi.tsx

```diff
@@ -22,5 +28,18 @@ export const navigateApi = (
     return;
   }
 
-  navigate(`/${matchingScreen.name.toLowerCase()}`);
+  // set additional inputs
+  const inputs = !isString(targetScreen) && data ? cloneDeep(data) : {};
```

Comment:
@sagardspeed2 i am not sure what syntax you are gonna support but i was looking if we can use 
```js
ensemble.navigateScreen("{name: 'home', inputs: {id: 4}}");
```
@evshi thoughts ?

Reply:
@sibteali786 currently this syntax will support the inputs.
```
ensemble.navigateScreen('screenName', payload)
```

@sibteali786 @evshi your thoughts on this ?

Reply:
It should follow the existing convention set by flutter side. Please check existing schema or docs if you're unsure https://docs.ensembleui.com/#/actions/8-navigateScreen?id=propertiesoptions

```yaml
      - Button:
          label: Navigate and pass inputs using code
          onTap:
            executeCode:
              body: |
                //@code
                ensemble.navigateScreen({
                  "name": "Action: navigateScreen inputs demo",
                  "inputs":  { 
                    "message": messageInput.value
                  }
                });

```

Reply:
okay, will update this as per the flutter schema